### ISSUE TITLE
EWSTimeZone.__eq__ should better accommodate other timezone implementations

### ISSUE THREAD

**Issue by earlchew** [ecederstrand/exchangelib#557]
`EWSTimeZone.__eq__` assumes that `other` is also an `ESWTimeZone` instance, but in an application with a diverse runtime environment, that might not be the case.

```
    def __eq__(self, other):
        return self.ms_id == other.ms_id
```

**Comment by ecederstrand** [comment]
How would you define equality of an `EWSTimeZone` instance and non-`EWSTimeZone` instance? Should we just return `False` always? If not, which other timezone implementations do you want to see supported? And do we match by a best-effort translation to region/location, or a full match of UTC offset and all future and historical DST transitions?

**Comment by earlchew** [comment]
I've suggested an approach in the pull request.

https://stackoverflow.com/questions/40780004/returning-notimplemented-from-eq

**Comment by earlchew** [comment]
I should add that returning `NotImplemented` solves the scenario that I encountered.

**Comment by ecederstrand** [comment]
Fixed in #561
